packages/kotti-ui/source/kotti-table/table/hooks.ts

```diff
@@ -0,0 +1,652 @@
+			}))
+		},
+		set(value) {
+			return (
+				value
+					// Remove entries that do not have a valid column
+					.filter((sorting) => columnIdSet.value.has(sorting.id))
+					.map((sorting) => ({
+						desc: sorting.value === 'descending',
+						id: sorting.id,
+					}))
+			)
+		},
+		value: ref([]),
+	})
+
+	const hiddenColumns = useComputedRef<VisibilityState, Set<COLUMN_ID>>({
+		get: (value) => {
+			const result = new Set<COLUMN_ID>()
+
+			for (const id of columnIdSet.value) {
+				if (value[id] === false) result.add(id)
+			}
+
+			return result
+		},
+		set: (newSet) => {
+			const newVisibilityState: VisibilityState = {}
+
+			for (const id of columnIdSet.value) {
+				newVisibilityState[id] = !newSet.has(id)
+			}
+
+			return newVisibilityState
+		},
+		value: ref(
+			Object.fromEntries(
+				params.value.columns.map((column) => [column.id, true]),
+			),
+		),
+	})
+
+	const triggerExpand = (rowId: string) => {
+		switch (params.value.expandMode) {
+			case null: {
+				expandedRows.value = {}
+				break
+			}
+			case 'single': {
+				// eslint-disable-next-line unicorn/prefer-ternary
+				if (rowId in expandedRows.value) {
+					expandedRows.value = {}
+				} else {
+					expandedRows.value = { [rowId]: true }
+				}
+				break
+			}
+			case 'multi': {
+				// eslint-disable-next-line unicorn/prefer-ternary
+				if (rowId in expandedRows.value) {
+					expandedRows.value = { ...expandedRows.value, [rowId]: false }
+				} else {
+					expandedRows.value = { ...expandedRows.value, [rowId]: true }
+				}
+			}
+		}
+	}
+
+	const draggedColumnId: Ref<COLUMN_ID | null> = ref(null)
+	/**
+	 * Primarily used for displaying the drop indicator line
+	 */
+	const dropTargetColumnIndex = ref<number | null>(null)
+	const successfullyDroppedColumnId = ref<string | null>(null)
+
+	const getMovedColumnOrder = (
+		fromColumnId: COLUMN_ID,
+		toIndex: number,
+	): COLUMN_ID[] => {
+		const fromIndex = columnOrder.value.indexOf(fromColumnId)
+		if (fromIndex === -1)
+			throw new Error(`Could not find column id ${fromColumnId}`)
+
+		const newOrder = columnOrder.value.toSpliced(fromIndex, 1)
+		newOrder.splice(
+			toIndex > fromIndex ? toIndex - 1 : toIndex,
+			0,
+			fromColumnId,
+		)
+		return newOrder
+	}
+
+	const table = useVueTable<ROW>(
+		computed(() => ({
+			autoResetExpanded: true,
+			columns: [
+				...(params.value.expandMode !== null
+					? [
+							columnHelper.display({
+								cell: ({ row }: CellContext<ROW, unknown>) => {
+									const rowBehavior = params.value.getRowBehavior({
+										row: row.original,
+										rowIndex: row.index,
+									})
+									const isDisabled = rowBehavior.disable?.expand ?? false
+
+									return h(
+										'div',
+										{
+											class: {
+												'kt-table-expand': true,
+												yoco: true,
+											},
+											domProps: {
+												ariaDisabled: String(isDisabled),
+												ariaExpanded: String(row.getIsExpanded()),
```

Comment:
Might need to be this

```suggestion
												'aria-expanded': String(row.getIsExpanded()),
```